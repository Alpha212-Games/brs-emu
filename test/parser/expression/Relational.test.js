const Parser = require("../../../lib/parser");
const { Lexeme } = require("../../../lib/Lexeme");
const BrsError = require("../../../lib/Error");

const { token, EOF } = require("../ParserTests");

describe("parser", () => {
    afterEach(() => BrsError.reset());

    describe("relational expressions", () => {
        it("parses less-than expressions", () => {
            let parsed = Parser.parse([
                token(Lexeme.Integer, 5),
                token(Lexeme.Less),
                token(Lexeme.Integer, 2),
                EOF
            ]);
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        it("parses less-than-or-equal-to expressions", () => {
            let parsed = Parser.parse([
                token(Lexeme.Integer, 5),
                token(Lexeme.LessEqual),
                token(Lexeme.Integer, 2),
                EOF
            ]);
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        it("parses greater-than expressions", () => {
            let parsed = Parser.parse([
                token(Lexeme.Integer, 5),
                token(Lexeme.Greater),
                token(Lexeme.Integer, 2),
                EOF
            ]);
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        it("parses greater-than-or-equal-to expressions", () => {
            let parsed = Parser.parse([
                token(Lexeme.Integer, 5),
                token(Lexeme.GreaterEqual),
                token(Lexeme.Integer, 2),
                EOF
            ]);
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        it("parses equality expressions", () => {
            let parsed = Parser.parse([
                token(Lexeme.Integer, 5),
                token(Lexeme.Equal),
                token(Lexeme.Integer, 2),
                EOF
            ]);
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });

        it("parses inequality expressions", () => {
            let parsed = Parser.parse([
                token(Lexeme.Integer, 5),
                token(Lexeme.LessGreater),
                token(Lexeme.Integer, 2),
                EOF
            ]);
            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });
    });
});