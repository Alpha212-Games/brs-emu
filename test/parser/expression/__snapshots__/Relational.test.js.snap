// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser relational expressions parses equality expressions 1`] = `
Array [
  Assignment {
    "name": Object {
      "kind": 21,
      "line": 1,
      "text": "_",
    },
    "value": Binary {
      "left": Literal {
        "value": Int32 {
          "kind": 3,
          "value": 5,
        },
      },
      "right": Literal {
        "value": Int32 {
          "kind": 3,
          "value": 2,
        },
      },
      "token": Object {
        "kind": 19,
        "line": 1,
        "text": "=",
      },
    },
  },
]
`;

exports[`parser relational expressions parses greater-than expressions 1`] = `
Array [
  Assignment {
    "name": Object {
      "kind": 21,
      "line": 1,
      "text": "_",
    },
    "value": Binary {
      "left": Literal {
        "value": Int32 {
          "kind": 3,
          "value": 5,
        },
      },
      "right": Literal {
        "value": Int32 {
          "kind": 3,
          "value": 2,
        },
      },
      "token": Object {
        "kind": 17,
        "line": 1,
        "text": ">",
      },
    },
  },
]
`;

exports[`parser relational expressions parses greater-than-or-equal-to expressions 1`] = `
Array [
  Assignment {
    "name": Object {
      "kind": 21,
      "line": 1,
      "text": "_",
    },
    "value": Binary {
      "left": Literal {
        "value": Int32 {
          "kind": 3,
          "value": 5,
        },
      },
      "right": Literal {
        "value": Int32 {
          "kind": 3,
          "value": 2,
        },
      },
      "token": Object {
        "kind": 18,
        "line": 1,
        "text": ">=",
      },
    },
  },
]
`;

exports[`parser relational expressions parses inequality expressions 1`] = `
Array [
  Assignment {
    "name": Object {
      "kind": 21,
      "line": 1,
      "text": "_",
    },
    "value": Binary {
      "left": Literal {
        "value": Int32 {
          "kind": 3,
          "value": 5,
        },
      },
      "right": Literal {
        "value": Int32 {
          "kind": 3,
          "value": 2,
        },
      },
      "token": Object {
        "kind": 20,
        "line": 1,
        "text": "<>",
      },
    },
  },
]
`;

exports[`parser relational expressions parses less-than expressions 1`] = `
Array [
  Assignment {
    "name": Object {
      "kind": 21,
      "line": 1,
      "text": "_",
    },
    "value": Binary {
      "left": Literal {
        "value": Int32 {
          "kind": 3,
          "value": 5,
        },
      },
      "right": Literal {
        "value": Int32 {
          "kind": 3,
          "value": 2,
        },
      },
      "token": Object {
        "kind": 15,
        "line": 1,
        "text": "<",
      },
    },
  },
]
`;

exports[`parser relational expressions parses less-than-or-equal-to expressions 1`] = `
Array [
  Assignment {
    "name": Object {
      "kind": 21,
      "line": 1,
      "text": "_",
    },
    "value": Binary {
      "left": Literal {
        "value": Int32 {
          "kind": 3,
          "value": 5,
        },
      },
      "right": Literal {
        "value": Int32 {
          "kind": 3,
          "value": 2,
        },
      },
      "token": Object {
        "kind": 16,
        "line": 1,
        "text": "<=",
      },
    },
  },
]
`;
